### DISCUSSION TITLE
Self-Hosted Gateway Fetching APIs from Both Dev and Production APIM Environments

### DISCUSSION THREAD

**Discussion by meezankhan9** [Azure/api-management-self-hosted-gateway#308] [General]
Hi all,

We are facing an issue with our self-hosted gateway configuration in Azure API Management. The setup involves a custom domain dev.configuration.api.xyz.com in our Dev APIM environment.

Here’s the situation:

We have configured the self-hosted gateway to work with Dev APIM, specifically using the Dev gateway apim-gw-aws-dev.

However, when connecting to the configuration endpoint dev.configuration.api.xyz.com, we’re seeing not only the Dev APIs but also the routes from our Production APIM (api.xyz.com), which should not be the case.

Expected behavior: It should only return APIs from the Dev gateway (apim-gw-aws-dev).

Microsoft’s Investigation:

We worked with Microsoft on the initial authentication issue, and the connectivity to the Dev APIM has been resolved. Below is an excerpt from the analysis:

“We observed an authorization failure due to an invalid token and attempted to generate a new one. When that didn’t resolve the issue, we switched the authorization to an app registration (ccc_apim-gw-aws-we-dev) as API Management Gateway Configuration Reader Role, which resolved the authentication problem. We are now able to retrieve all Dev APIs as expected."

However, when connecting via the dev.configuration.api.xyz.com endpoint, we are still seeing APIs from the Production environment.

Microsoft’s team has advised reaching out to the GitHub community for further assistance on this issue.

What We Need Help With:

Has anyone encountered this issue where a self-hosted gateway is fetching APIs from both Dev and Production when it should only be fetching from the specified Dev gateway? If so, what could be the cause and solution to restrict the gateway to only return the Dev environment APIs?

Any guidance or pointers would be appreciated!

Thanks,
Meezan

**Comment by tomkerkhove** [comment]
There is no way SHGW can have configuration from 2 API Management instances. Maybe you have tested with the production instance and the configuration is in the local backup which you can delete or the custom DNS was assigned to production in the past.

If you need further help, please open a support ticket as we cannot access anything without a ticket.

**Reply by meezankhan9** [reply]
Hi tom 

Thanks for the details we diagnosed and found below details

There is no connectivity between Production and Development in part of IAM or some other gateway configuration.
 
ccc_apim-gw-aws-we-dev is assigned to ate_ccc_ccc not ane_ccc_ccc
 
We also looked in the document and seen dns mapped to default domain, but we are using application gateway, so we have connected to the traffic manager dns instead of direct apim dns
